go.mod: bump datadriven

RaduBerinde opened the pull request:
Bump datadriven to incorporate a fix
(https://github.com/cockroachdb/datadriven/pull/60).

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154867)
<!-- Reviewable:end -->

RaduBerinde commented:
TFTR!

Remaining unit test failure is a flake (https://github.com/cockroachdb/cockroach/issues/154805)

bors r+

kev-cao commented:
@RaduBerinde We'll also want to backport this to 25.4 — https://github.com/cockroachdb/cockroach/issues/154833

RaduBerinde committed:
go.mod: bump datadriven

Bump datadriven to incorporate a fix
(https://github.com/cockroachdb/datadriven/pull/60).

Epic: none
Release note: None
